src/ogc-api/endpoint.ts

```diff
@@ -252,19 +263,38 @@ ${e.message}`);
     ]).then(checkHasRecords);
   }
 
+  /**
+   * A Promise which resolves to a boolean indicating whether the endpoint offers environmental data retrieval (EDR) queries.
+   */
+  get hasEnvironmentalDataRetrieval(): Promise<boolean> {
+    return Promise.all([this.conformanceClasses]).then(
+      checkHasEnvironmentalDataRetrieval
+    );
+  }
+
+  /*
+   * A Promise which resolves to a class for constructing EDR queries
+   */
+  public async edr(collection_id: string): Promise<EDRQueryBuilder> {
+    const collection = await this.getCollectionDocument(collection_id);
+    return new EDRQueryBuilder(collection);
```

Comment:
It'd be great if you could include a caching mechanism here to avoid recreating the `EdrQueryBuilder` instance every time, like for https://github.com/camptocamp/ogc-client/blob/961fd4579f577dad10746fed5c282092058584d6/src/ogc-api/endpoint.ts#L65